tools/proteomiqon_psmbasedquantification/proteomiqon_psmbasedquantification.xml

```diff
@@ -0,0 +1,449 @@
+                                    "Fields": [
+                                        ${XicExtraction.XicProcessing.ProcessingTypeCond.MinPeakLengthCond.MinPeakLength}
+                                    ]
+                                },
+                                #else
+                                "MinPeakLength": null,
+                                #end if
+                                "MaxPeakLength": ${XicExtraction.XicProcessing.ProcessingTypeCond.MaxPeakLength},
+                                "NoiseQuantile": ${XicExtraction.XicProcessing.ProcessingTypeCond.NoiseQuantile},
+                                "MinSNR": ${XicExtraction.XicProcessing.ProcessingTypeCond.MinSNR},
+                            }
+                        ]
+                        #else
+                        "Case": "SecondDerivative",
+                        "Fields": [
+                            {
+                                "MinSNR": ${XicExtraction.XicProcessing.ProcessingTypeCond.MinSNR},
+                                "PolynomOrder": ${XicExtraction.XicProcessing.ProcessingTypeCond.PolynomOrder},
+                                "WindowSize": {
+                                    "Case": "${XicExtraction.XicProcessing.ProcessingTypeCond.WindowSizeCond.WindowSizeType}",
+                                    "Fields" : [
+                                        ${XicExtraction.XicProcessing.ProcessingTypeCond.WindowSizeCond.WindowSize}                 
+                                    ]
+                                }
+                            }
+                        ]    
+                        #end if
+                    },
+                    #if $XicExtraction.TopKPSMsCond.TopKPSMsEnabled == "yes"
+                    "TopKPSMs": {
+                        "Case": "Some",
+                        "Fields": [
+                            ${XicExtraction.TopKPSMsCond.TopKPSMs}
+                        ]
+                    }
+                    #else
+                    "TopKPSMs": null
+                    #end if
+                },
+                #if $BaseLineCorrectionCond.BaseLineCorrectionEnabled == "yes"
+                "BaseLineCorrection": {
+                    "Case": "Some",
+                    "Fields": [
+                        {
+                            "MaxIterations":$BaseLineCorrectionCond.MaxIterations,
+                            "Lambda":$BaseLineCorrectionCond.Lambda,
+                            "P":$BaseLineCorrectionCond.P
+                        }
+                    ]
+                }
+                #else
+                "BaseLineCorrection": null
+                #end if
+            }
+            ]]>
+        </configfile>
+    </configfiles>
+    <inputs>
+        <param name="instrumentOutput" type="data" format="sqlite" label="Instrument output" help="Specify mass spectrometry data you want to analyze."/>
```

Comment:
Just a hint, the sqlite datatype does support metadata, so you can say here "I only take sqlite dbs with a column count of X". Like here: https://github.com/galaxyproject/tools-iuc/blob/e22821f51ae326b3696a9456cbc44f9982512a52/tools/gemini/gemini_macros.xml#L46

Reply:
Tell me if I am missing something, but would I not have to do this  specific task with a validator as opposed to options+filters? https://docs.galaxyproject.org/en/latest/dev/schema.html#validators-for-data-and-data-collection-parameters